libs/src/api/user.js

```diff
@@ -291,27 +299,39 @@ class Users extends Base {
     const newMetadata = this._cleanUserMetadata(metadata)
     this._validateUserMetadata(newMetadata)
 
+    // Error if libs instance does not already have existing user state
     const user = this.userStateManager.getCurrentUser()
     if (!user) {
       throw new Error('No current user')
     }
+
+    // Preserve old metadata object
     const oldMetadata = { ...user }
 
-    // Upload new metadata
-    const { metadataMultihash, metadataFileUUID } = await this.creatorNode.uploadCreatorContent(
-      newMetadata
-    )
-    // Update new metadata on chain
-    let updatedMultihashDecoded = Utils.decodeMultihash(metadataMultihash)
-    const { txReceipt } = await this.contracts.UserFactoryClient.updateMultihash(userId, updatedMultihashDecoded.digest)
-    const { latestBlockNumber } = await this._updateUserOperations(newMetadata, oldMetadata, userId)
+    // Update user creator_node_endpoint on chain if applicable
     if (newMetadata.creator_node_endpoint !== oldMetadata.creator_node_endpoint) {
-      await this._waitForCreatorNodeUpdate(newMetadata.user_id, newMetadata.creator_node_endpoint)
+      await this.contracts.UserFactoryClient.updateCreatorNodeEndpoint(userId, newMetadata['creator_node_endpoint'])
+
+      // Ensure DN has indexed creator_node_endpoint change
+      await this._waitForCreatorNodeEndpointIndexing(newMetadata.user_id, newMetadata.creator_node_endpoint)
```

Comment:
Looks good to me. Do we need to double check here that we're "connected" to the right creator node now? this.creatorNode.uploadCreatorContent would upload to the old endpoint i think

Reply:
i think since the only place we allow the user to manually change the replica set is the service selection modal and the saga calls `setCreatorNodeEndpoint` before calling `updateCreator` (https://github.com/AudiusProject/audius-client/blob/master/src/containers/service-selection/store/sagas.ts#L132) it updates the user's state in libs to log into the right now. same with the rolloverNode sanity check

with that being said, i think having logic in this if statement to set the creator node endpoint and confirm the endpoint if the metadata is being updated would be a good thing @SidSethi @raymondjacobson